### DISCUSSION TITLE
bug: LazyVim can't clone multiple plugins at first launch on RaspberryPi Zero

### DISCUSSION THREAD

**Discussion by iharkatkavets** [LazyVim/LazyVim#4396] [Q&A] [answered]
### Did you check docs and existing issues?

- [X] I have read all the LazyVim docs
- [X] I have updated the plugin to the latest version before submitting this issue
- [X] I have searched the existing issues of LazyVim
- [X] I have searched the existing issues of plugins related to this issue

### Neovim version (nvim -v)

NVIM v0.11.0-dev-700+gbcae8be91

### Operating system/version

Linux pizero 6.6.31+rpt-rpi-v6 #1 Raspbian 1:6.6.31-1+rpt1 (2024-05-29) armv6l GNU/Linux

### Describe the bug

LazyVim can't fetch(clone) plugins at first launch

<img width="1677" alt="Screenshot 2024-09-03 at 10 36 32" src="https://github.com/user-attachments/assets/2b1da4a7-18c0-4e34-b04a-000a3557ef44">
<img width="1728" alt="Screenshot 2024-09-03 at 10 37 07" src="https://github.com/user-attachments/assets/68852398-c704-4c78-bec2-9d407894a4be">
<img width="1714" alt="image" src="https://github.com/user-attachments/assets/0948f313-1508-45f5-9433-ccd6d3f5d578">

### Steps To Reproduce

1. Compile neovim from sources 
2. Add minimal `init.lua` config
4. Launch neovim 

### Expected Behavior

All plugins download at first launch

### Repro

```lua
vim.env.LAZY_STDPATH = ".repro"
load(vim.fn.system("curl -s https://raw.githubusercontent.com/folke/lazy.nvim/main/bootstrap.lua"))()

require("lazy.minit").repro({
  spec = {
    { "LazyVim/LazyVim", import = "lazyvim.plugins" },
    -- add any other plugins here
  },
})
```

**Comment by dpetka2001** [comment]
What does that have to do with Lazyvim? This is the package manager's responsibility and your git configuration. Check your git version and configuration. Preferably try to disable your git configuration and try again.

**Comment by iharkatkavets** [comment]
> What does that have to do with Lazyvim? This is the package manager's responsibility and your git configuration. Check your git version and configuration. Preferably try to disable your git configuration and try again.

`git version 2.39.2`

But as you see, LazyVim itself is able to be cloned, but everything after is not .
I also can clone any project from GitHub via HTTPS or SSL without issues.

I can't get more logs from LazyVim even by set debug level to 'trace'. I also assume that the issue is related with compiled `neovim` build, but I don't know how to check that

**Comment by dpetka2001** [comment]
What do you have in your `~/.config/git/config`?

**Comment by iharkatkavets** [comment]
> What do you have in your `~/.config/git/config`?

I don't have the file and `~/.config/git` directory at all

**Comment by iharkatkavets** [comment]
Again, `lazyvim` can't download `bufferline.nvim` plugin, but manually I can 

<img width="1651" alt="image" src="https://github.com/user-attachments/assets/df572c09-99d3-4121-8d7d-bb27977558d4">
<img width="1718" alt="image" src="https://github.com/user-attachments/assets/a618a990-0e7c-4175-b426-3460cc2299cf">

**Reply by iharkatkavets** [reply]
> Again, `lazyvim` can't download `bufferline.nvim` plugin, but manually I can
> 
> <img alt="image" width="1651" src="https://private-user-images.githubusercontent.com/1083951/364041618-df572c09-99d3-4121-8d7d-bb27977558d4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUzNzY5MTEsIm5iZiI6MTcyNTM3NjYxMSwicGF0aCI6Ii8xMDgzOTUxLzM2NDA0MTYxOC1kZjU3MmMwOS05OWQzLTQxMjEtOGQ3ZC1iYjI3OTc3NTU4ZDQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkwMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MDNUMTUxNjUxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NGQ3OTE3OTdhMTM2NDQzNzNkNGNhYTYzYzdjY2VlZGNjZTI4NDliZDdlNDUwNTVkOTA5Y2ZiZWQzM2MxMjBjOSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.1tBI-tOOu-HajsecNphkOvyRYzWKEmG32jVu76GyjI8"> <img alt="image" width="1718" src="https://private-user-images.githubusercontent.com/1083951/364041995-a618a990-0e7c-4175-b426-3460cc2299cf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUzNzY5MTEsIm5iZiI6MTcyNTM3NjYxMSwicGF0aCI6Ii8xMDgzOTUxLzM2NDA0MTk5NS1hNjE4YTk5MC0wZTdjLTQxNzUtYjQyNi0zNDYwY2MyMjk5Y2YucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkwMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MDNUMTUxNjUxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MGIyNGZlNjliZGYzODVjYzQzYTI1MDljMWE4NjFmYWFlOGEyZDNjMGQ2Y2NkZTI0NzA5NTg2NmMzNDcxMzc0MyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.gVyQxGD0Dw9ZygEiN664ba5UBLJ65HJL0qAHQ_33dv0">

Is there a way to limit "clone queue" to some number? I think it might be possible that too many plugins have been requested for downloading at once, but RaspberryPI Zero is quite slow device

**Reply by iharkatkavets** [reply]
> > Again, `lazyvim` can't download `bufferline.nvim` plugin, but manually I can
> > <img alt="image" width="1651" src="https://private-user-images.githubusercontent.com/1083951/364041618-df572c09-99d3-4121-8d7d-bb27977558d4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUzNzY5MTEsIm5iZiI6MTcyNTM3NjYxMSwicGF0aCI6Ii8xMDgzOTUxLzM2NDA0MTYxOC1kZjU3MmMwOS05OWQzLTQxMjEtOGQ3ZC1iYjI3OTc3NTU4ZDQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkwMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MDNUMTUxNjUxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NGQ3OTE3OTdhMTM2NDQzNzNkNGNhYTYzYzdjY2VlZGNjZTI4NDliZDdlNDUwNTVkOTA5Y2ZiZWQzM2MxMjBjOSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.1tBI-tOOu-HajsecNphkOvyRYzWKEmG32jVu76GyjI8"> <img alt="image" width="1718" src="https://private-user-images.githubusercontent.com/1083951/364041995-a618a990-0e7c-4175-b426-3460cc2299cf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUzNzY5MTEsIm5iZiI6MTcyNTM3NjYxMSwicGF0aCI6Ii8xMDgzOTUxLzM2NDA0MTk5NS1hNjE4YTk5MC0wZTdjLTQxNzUtYjQyNi0zNDYwY2MyMjk5Y2YucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkwMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MDNUMTUxNjUxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MGIyNGZlNjliZGYzODVjYzQzYTI1MDljMWE4NjFmYWFlOGEyZDNjMGQ2Y2NkZTI0NzA5NTg2NmMzNDcxMzc0MyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.gVyQxGD0Dw9ZygEiN664ba5UBLJ65HJL0qAHQ_33dv0">
> 
> Is there a way to limit "clone queue" to some number? I think it might be possible that too many plugins have been requested for downloading at once, but RaspberryPI Zero is quite slow device

YES! I reduced the number of plugins in the config - and they have been downloaded!
<img width="1446" alt="image" src="https://github.com/user-attachments/assets/42ed0e01-faa9-417b-af2e-3aad7a12c3cd">

**Reply by iharkatkavets** [reply]
So to prove the theory above I need to find a config parameter that adjust number of simultaneous "clone" operations by lazyvim.
@dpetka2001 Could you help me with this ?

**Reply by dpetka2001** [reply, marked as answer]
Then you might be interested in `lazy.nvim`'s [configuration](https://lazy.folke.io/configuration). There's a key `concurrency`. Try setting that to `1` in your `~/.config/nvim/lua/config/lazy.lua` file. If that works try increasing that number until it breaks again to find the sweet spot.

**Reply by iharkatkavets** [reply]
Thank you, the max `concurrency` parameter that allows to download plugins on rpi zero is `10`

**Reply by chavesdaniel** [reply]
I'm having the same problem, but in Windows 11:

![image](https://github.com/user-attachments/assets/e90abde4-febd-44b8-9d6e-4910255822e0)

then:

![image](https://github.com/user-attachments/assets/8cd5706a-c976-4565-8983-16e0e019c786)

**Reply by baifangkual** [reply]
@chavesdaniel 
did you solve the problem you encountered? I'm having the same issue on Windows too
and I can't get concurrency to work

**Reply by manuelarte** [reply]
I solved it by `  concurrency = 1,`
